source/simulator/solver.cc

```diff
@@ -808,16 +808,32 @@ namespace aspect
 
                 f.close();
 
+                // avoid a deadlock that was fixed after deal.II 8.5.0
+#if DEAL_II_VERSION_GTE(9,0,0)
                 AssertThrow (false,
                              ExcMessage (std::string("The iterative Stokes solver "
                                                      "did not converge. It reported the following error:\n\n")
                                          +
                                          exc.what()
                                          + "\n See " + parameters.output_directory+"solver_history.txt"
                                          + " for convergence history."));
+#else
+                std::cerr << "The iterative Stokes solver "
+                          << "did not converge. It reported the following error:\n\n"
+                          << exc.what()
+                          << "\n See "
+                          << parameters.output_directory
+                          << "solver_history.txt for convergence history."
+                          << std::endl;
+#endif
               }
             else
-              throw QuietException();
+              {
+#if DEAL_II_VERSION_GTE(9,0,0)
+                throw QuietException();
+#endif
+              }
+            std::abort();
```

Comment:
move this into the `#else` above and into a new `#else` in the second conditional. This way we will remember to remove it when we finally require 9.0.